### PR TITLE
fixed libc command on mingw

### PR DESCRIPTION
**Pull request by emekoi**
zig libc command now generates a libc file correctly. compiling with libc is still broken though.

### CODE REVIEW

File: src/libc_installation.cpp [hunk 1]
```diff
@@ -203,6 +203,130 @@ static Error zig_libc_find_native_msvc_include_dir(ZigLibCInstallation *self, Zi
...
+        os_path_dirname(out_stdout, out);
+    } else {
+        buf_init_from_buf(out, out_stdout);
+    }
+    return ErrorNone;
+}
+static Error zig_libc_find_native_crt_dir_gnu(ZigLibCInstallation *self, bool verbose) {
+    return zig_libc_cc_print_file_name("crt1.o", &self->crt_dir, true, verbose);
+}
+static Error zig_libc_find_native_include_dir_gnu(ZigLibCInstallation *self, bool verbose) {
```
Comment (andrewrk) [inline src/libc_installation.cpp]: Is this function really so different than `zig_libc_find_native_include_dir_posix` that it needs to be a different implementation? Same question for `zig_libc_cc_print_file_name` and the other corresponding implementation.

File: src/libc_installation.cpp [hunk 2]
```diff
@@ -111,7 +111,7 @@ Error zig_libc_parse(ZigLibCInstallation *libc, Buf *libc_file, const ZigTarget
     }
 
     if (buf_len(&libc->msvc_lib_dir) == 0) {
-        if (target->os == OsWindows) {
+        if (target->os == OsWindows && target->abi == ZigLLVM_MSVC) {
```
Comment (andrewrk) [inline src/libc_installation.cpp]: For MinGW/MSVC detection, can we do it with this logic?

```c++
bool is_mingw(ZigTarget *target) {
    return target->os == OsWindows and target_abi_is_gnu(target);
}
```

And then if it's not mingw, we assume MSVC.

Comment (emekoi) [reply src/libc_installation.cpp]: i think only the `target_abi_is_gnu(target->abi)` is necessary as that would fit better in to more places such as `zig_libc_parse` and `zig_libc_find_native`.

File: src/libc_installation.cpp (hunk 1 above)
Comment (emekoi) [reply src/libc_installation.cpp]: probably not. a couple `#ifdef` would probably be enough to make the original function work on both posix and mingw systems.

**Comment by andrewrk** [comment]
Thanks for your patience on this one @emekoi. Nice work

**Comment by emekoi** [comment]
no problem.

**Pull request merged**